src/Shared/Grpc/EntityConversions.cs

```diff
@@ -0,0 +1,443 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using DurableTask.Core.Entities;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.DurableTask;
+using Newtonsoft.Json;
+using DTCore = DurableTask.Core;
+using P = Microsoft.DurableTask.Protobuf;
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Utilities for converting between representations of entity history events. The older backends represent entity
+/// messages as external events, using a JSON encoding defined in DT Core.
+/// Starting with the DTS backend, we use explicit, separate
+/// protobuf encodings for all entity-related history events.
+/// </summary>
+static class EntityConversions
+{
+    // we copied the relevant data members from DT.Core to allow us to convert between the data structures
+    // used in the backend, and the legacy encoding of entities within orchestration histories.
+
+    // The point of this class is to reverse a Newtonsoft serialization that happened in prior DT code.
+    // To do this reliably we use the same Newtonsoft.
+    // This is not introducing a new dependency, and should be eliminated once the original dependency is eliminated.
```

Comment:
Just to be clear, if we remove the DT.Core dependency from this SDK, will that also allow us to remove this dependency on Newtonsoft.Json?

Reply:
Yes. The point of this class is to "undo Newtonsoft serialization" that was unfortunately already done inside DT.Core. So the Newtonsoft dependency goes away when either (a) Newtonsoft dependency is removed from DT.Core, or (b) DT.Core dependency is removed.